ee/tabby-ui/app/(dashboard)/profile/components/avatar.tsx

```diff
@@ -1,17 +1,113 @@
-import NiceAvatar, { genConfig } from 'react-nice-avatar'
+'use client'
 
+import { ChangeEvent, useState } from 'react'
+import { toast } from 'sonner'
+
+import { graphql } from '@/lib/gql/generates'
 import { useMe } from '@/lib/hooks/use-me'
+import { useMutation } from '@/lib/tabby/gql'
+import { delay } from '@/lib/utils'
+import { Button } from '@/components/ui/button'
+import { IconCloudUpload, IconSpinner } from '@/components/ui/icons'
+import { Separator } from '@/components/ui/separator'
+import { mutateAvatar, UserAvatar } from '@/components/user-avatar'
+
+const uploadUserAvatarMutation = graphql(/* GraphQL */ `
+  mutation uploadUserAvatarBase64($id: ID!, $avatarBase64: String!) {
+    uploadUserAvatarBase64(id: $id, avatarBase64: $avatarBase64)
+  }
+`)
 
 export const Avatar = () => {
+  const [isSubmitting, setIsSubmitting] = useState(false)
+  const [uploadedImgString, setUploadedImgString] = useState('')
   const [{ data }] = useMe()
-
+  const uploadUserAvatar = useMutation(uploadUserAvatarMutation)
   if (!data?.me?.email) return null
 
-  const config = genConfig(data?.me?.email)
+  const onPreviewAvatar = (e: ChangeEvent<HTMLInputElement>) => {
+    const file = e.target.files ? e.target.files[0] : null
+    if (file) {
+      const reader = new FileReader()
+
+      reader.onloadend = () => {
+        const imageString = reader.result as string
+        setUploadedImgString(imageString)
+      }
+
+      reader.readAsDataURL(file)
+    }
+  }
+
+  const onUploadAvatar = async () => {
+    setIsSubmitting(true)
+    try {
+      const mimeHeaderMatcher = new RegExp('^data:image/.+;base64,')
+      const avatarBase64 = uploadedImgString.replace(mimeHeaderMatcher, '')
```

Comment:
should be better library / approach to handle the parsing part?

Reply:
Did find good recommendation from gpt
its method is more straight forward
```js
// Get the result from the FileReader
const base64Data = reader.result;

// Split the string on the comma to remove the MIME type information
const base64String = base64Data.split(',')[1];
```
I would take this `split` method